efficientdet/anchors.py

```diff
@@ -183,6 +216,75 @@ def _generate_anchor_boxes(image_size, anchor_scale, anchor_configs):
   anchor_boxes = np.vstack(boxes_all)
   return anchor_boxes
 
+def _generate_detections_tf(cls_outputs, box_outputs, anchor_boxes, indices,
+                         classes, image_id, image_scale, num_classes, use_native_nms=False):
+  """Generates detections with RetinaNet model outputs and anchors.
+
+  Args:
+    cls_outputs: a numpy array with shape [N, 1], which has the highest class
+      scores on all feature levels. The N is the number of selected
+      top-K total anchors on all levels.  (k being MAX_DETECTION_POINTS)
+    box_outputs: a numpy array with shape [N, 4], which stacks box regression
+      outputs on all feature levels. The N is the number of selected top-k
+      total anchors on all levels. (k being MAX_DETECTION_POINTS)
+    anchor_boxes: a numpy array with shape [N, 4], which stacks anchors on all
+      feature levels. The N is the number of selected top-k total anchors on
+      all levels.
+    indices: a numpy array with shape [N], which is the indices from top-k
+      selection.
+    classes: a numpy array with shape [N], which represents the class
+      prediction on all selected anchors from top-k selection.
+    image_id: an integer number to specify the image id.
+    image_scale: a float tensor representing the scale between original image
+      and input image for the detector. It is used to rescale detections for
+      evaluating with the original groundtruth annotations.
+    num_classes: a integer that indicates the number of classes.
+  Returns:
+    detections: detection results in a tensor with each row representing
+      [image_id, y, x, height, width, score, class]
+  """
+  anchor_boxes = tf.gather(anchor_boxes, indices)
+
+  scores = tf.math.sigmoid(cls_outputs)
+  # apply bounding box regression to anchors
+  boxes = decode_box_outputs_tf(
+      tf.transpose(box_outputs, [1, 0]), tf.transpose(anchor_boxes, [1, 0]))
+
+  def _else(detections):
+    boxes_cls = tf.gather(boxes, indices)
+    scores_cls = tf.gather(scores, indices)
+    # Select top-scoring boxes in each class and apply non-maximum suppression
+    # (nms) for boxes in the same class. The selected boxes from each class are
+    # then concatenated for the final detection outputs.
+    all_detections_cls = tf.concat([tf.reshape(boxes_cls, [-1, 4]), scores_cls], axis=1)
+    if use_native_nms:
+      top_detection_idx = tf.image.non_max_suppression(all_detections_cls[:, :4],
```

Comment:
This is great! Did you verify both use_native_nms=True and False work?

Reply:
Yes, both work well, but get different result.